### ISSUE TITLE
dwarf: missing arguments and return values

### ISSUE THREAD

**Issue by Taeung** [namhyung/uftrace#664]
After recording with `-a` option, I found missing arg/ret vaules of `fsync()` and `fileno()`.
Is this a problem ? I thought arg/ret values of `fsync@plt` and `fileno@plt` can be recorded.
```bash
$ uftrace replay
# DURATION     TID     FUNCTION
   0.479 us [  4772] | __monstartup();
   0.143 us [  4772] | __cxa_atexit();
            [  4772] | main() {
 681.082 us [  4772] |   fopen("hello.txt", "w") = 0x55683f8e1e70;
   5.584 us [  4772] |   fwrite(&_IO_stdin_used, 1, 6, 0x55683f8e1e70) = 6;
   0.237 us [  4772] |   fileno();
 914.574 us [  4772] |   fsync();
  11.751 us [  4772] |   printf("%d\n") = 2;
  38.833 us [  4772] |   fclose(0x55683f8e1e70) = 0;
   1.656 ms [  4772] | } /* main */
```
[pf.uftrace.data.tar.gz](https://github.com/namhyung/uftrace/files/2807064/pf.uftrace.data.tar.gz)

the source code of the target program:
```c
#include <unistd.h>
#include <stdio.h>
#include <stdlib.h>

void main()
{
	FILE *fp = fopen("hello.txt", "w");
	int ret;

	if (!fp)
		return;
	
	fprintf(fp, "hello\n");
	ret = fsync(fileno(fp));
	printf("%d\n", ret);
	fclose(fp);
}
```

**Comment by namhyung** [comment]
No, PLT calls don't use DWARF.  It's hard-coded in `misc/prototypes.h`.

**Comment by honggyukim** [comment]
@Taeung You can apply the following change for them:
```diff
diff --git a/misc/prototypes.h b/misc/prototypes.h
index c9bdb06..10a5db6 100644
--- a/misc/prototypes.h
+++ b/misc/prototypes.h
@@ -29,6 +29,9 @@ void *bsearch(const void *key, const void *base, size_t nmemb, size_t size, func

 void exit(int status);

+int fileno(FILE *stream);
+int fsync(int fd);
+
 ////////////////////////////////////////////////////////////////////////////////
 // memory
 void *malloc(size_t size);
diff --git a/utils/auto-args.h b/utils/auto-args.h
index 544f515..d4e616e 100644
--- a/utils/auto-args.h
+++ b/utils/auto-args.h
@@ -39,6 +39,8 @@ static char *auto_args_list =
        "qsort_r@arg1/x,arg2/u,arg3/u,arg4/p,arg5/x;"
        "bsearch@arg1/x,arg2/x,arg3/u,arg4/u,arg5/p;"
        "exit@arg1/d32;"
+       "fileno@arg1/x;"
+       "fsync@arg1/d32;"
        "malloc@arg1/u;"
        "free@arg1/x;"
        "calloc@arg1/u,arg2/u;"
@@ -208,6 +210,8 @@ static char *auto_retvals_list =
        "strtod@retval/f64;"
        "strtof@retval/f32;"
        "bsearch@retval/x;"
+       "fileno@retval/d32;"
+       "fsync@retval/d32;"
        "malloc@retval/x;"
        "calloc@retval/x;"
        "realloc@retval/x;"
```
Actually, `utils/auto-args.h` doesn't have to be modified because it's auto generated file based on `misc/prototypes.h`.

**Comment by Taeung** [comment]
Ah, I got confused.. all arg/return variable types of `@plt` can be gained.
@honggyukim thanks for the change !!
Close this ? @namhyung

**Comment by namhyung** [comment]
I don't think we can add every library functions for auto-args.  I'd like to keep mostly used and/or generally useful functions.  But I know it's hard to determine, will add the above anyway.

And as you know, you can add argspec for functions you know if it's missing like below

```
$ uftrace -A fsync@arg1  yourprog
```

**Comment by namhyung** [comment]
Merged: c071446
